### PR TITLE
Add logging recipe

### PR DESCRIPTION
**Pull request by mlazos**
Added torch logging recipe

### CODE REVIEW

**Comment by williamwen42** [comment]
Can we point to further logging documentation, or give a description for more of the logging options (e.g. recompiles, graph_breaks, source_trace, aot_graphs, etc.)?

File: recipes_source/torch_logs.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+==========================================================================================
+**Author:** `Michael Lazos <https://github.com/mlazos>`_
+"""
+
+import logging
+
+######################################################################
+#
+# This tutorial introduces the TORCH_LOGS environment variable and python API and
+# demonstrates how to apply it to observe the phases torch.compile.
```
Comment (svekars) [inline recipes_source/torch_logs.py:12]:
```suggestion
# This tutorial introduces the ``TORCH_LOGS`` environment variable, as well ass the Python API, and
# demonstrates how to apply it to observe the phases  of ``torch.compile``.
```

File: recipes_source/torch_logs.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+#
+#   This tutorial requires PyTorch 2.2.0 or later.
+#
+#
+
+
+######################################################################
+# Setup
+# ~~~~~~~~~~~~~~~~~~~~~
+# In this example we'll setup a simple python function which performs an elementwise
```
Comment (svekars) [inline recipes_source/torch_logs.py:24]:
```suggestion
# In this example, we'll set up a simple Python function which performs an elementwise
```

File: recipes_source/torch_logs.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+# add and observe the compilation process with TORCH_LOGS python API.
```
Comment (svekars) [inline recipes_source/torch_logs.py:25]:
```suggestion
# add and observe the compilation process with ``TORCH_LOGS`` Python API.
```

File: recipes_source/torch_logs.py [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+#
+# .. note::
+#
+#   There is also an environment variable TORCH_LOGS, which can be used to
```
Comment (svekars) [inline recipes_source/torch_logs.py:29]:
```suggestion
#   There is also an environment variable ``TORCH_LOGS``, which can be used to
```

File: recipes_source/torch_logs.py [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+torch._logging.set_logs(fusion=True)
+fn(*inputs)
+
+separator("Output Code")
+# View output code generated by inductor
+# TORCH_LOGS="output_code"
+torch._logging.set_logs(output_code=True)
+fn(*inputs)
+
+separator("")
```
Comment (svekars) [inline recipes_source/torch_logs.py:76]: Some sort of conclusion with links to more torch.compile / dynamo docs would be nice to have.

**Review by williamwen42 — APPROVED** [review]

**Review by svekars — APPROVED** [review]
Looks good! Will merge the week of Jan 2.

**Pull request merged**